### ISSUE TITLE
Style/HashAsLastArrayItem copy with style "no_braces" can result in invalid code

### ISSUE THREAD

**Issue by aried3r** [rubocop-hq/rubocop#8323]
## Expected behavior

Using `Style/HashAsLastArrayItem` with `EnforcedStyle: no_braces` should yield valid Ruby code.

I was planning to use `braces` but wanted to see which parts of our code base might use the `no_braces` style currently and wanted to see what would change with either of the two styles.

## Actual behavior

The cop is not safe and results in invalid Ruby code.

## Steps to reproduce the problem

```YAML
# .rubocop.yml
AllCops:
  NewCops: enable

Style/HashAsLastArrayItem:
  EnforcedStyle: no_braces
```

```Ruby
# hash_as_last_array_item.rb before
# frozen-string-literal: true

a = [
  {
    foo: :bar
  },
  {
    foo: :baz
  }
]

pp a
```

```Console
$ ruby hash_as_last_array_item.rb
[{:foo=>:bar}, {:foo=>:baz}]
```

```Console
$ rubocop -a hash_as_last_array_item.rb
Inspecting 1 file
E

Offenses:

hash_as_last_array_item.rb:4:3: C: [Corrected] Style/HashAsLastArrayItem: Omit the braces around the hash.
  { ...
  ^
hash_as_last_array_item.rb:6:3: E: Lint/Syntax: unexpected token tCOMMA
(Using Ruby 2.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  ,
  ^
hash_as_last_array_item.rb:7:3: C: [Corrected] Style/HashAsLastArrayItem: Omit the braces around the hash.
  { ...
  ^

1 file inspected, 3 offenses detected, 2 offenses corrected
```

```Ruby
# hash_as_last_array_item.rb after
# frozen-string-literal: true

a = [
  
    foo: :bar
  ,
  
    foo: :baz
  
]

pp a
```

```Console
$ ruby hash_as_last_array_item.rb
hash_as_last_array_item.rb:6: syntax error, unexpected ',', expecting ']'
hash_as_last_array_item.rb:10: syntax error, unexpected ']', expecting end-of-input
```

## RuboCop version

```
$ rubocop -V
0.88.0 (using Parser 2.7.1.4, rubocop-ast 0.1.0, running on ruby 2.7.1 x86_64-darwin19)
```
